pipeline/Makefile

```diff
@@ -0,0 +1,109 @@
+.DEFAULT_GOAL := help
+
+CONFIG_PATH := brca_pipeline_cfg.mk
+include $(CONFIG_PATH)
+
+checkout: ## checking out code base
+	[ -d $(CODE_BASE) ] || git clone $(BRCA_GIT_REPO) $(CODE_BASE)
+	cd $(CODE_BASE) && git checkout $(GIT_COMMIT)
+
+build-docker: ## build main pipeline docker image
+	cd $(CODE_BASE)/pipeline/docker && ./build_docker_image.sh $(GIT_COMMIT) $(RELEASE_TAG) || { echo "$(date) Something wrong with docker image creation"; exit 1; }
+
+download-resources: ## download resources files
+	$(CODE_BASE)/pipeline/download_resources_files.sh $(RESOURCES_DIR) >> "$(LOG_DIR)/download_resources_$(RELEASE_TAG).log" 2>&1 || { echo "Downloading resource files failed"; exit 1; }
+
+
+start-local-uta: ## starting local uta docker container
+	[ `docker ps -f name=$(UTA_CONTAINER) | wc -l` -gt 1 ] || docker run -dit --name $(UTA_CONTAINER) -p $(UTA_PORT):5432 $(UTA_DOCKER_IMAGE)
+
+.ONESHELL:
+setup-files: ## setup various directories to run pipeline
+	[ -f $(CREDENTIALS_PATH) ] || { touch $(CREDENTIALS_PATH); echo "WARNING: files $(CREDENTIALS_PATH) doesn't exist.Creating it"; }
+	[ -f $(RELEASE_NOTES_PATH) ] || { touch $(RELEASE_NOTES_PATH) ; echo "WARNING: files $(RELEASE_NOTES_PATH) doesn't exist. Creating it"; }
+	[ -d $(SYNAPSE_CACHE) ] || mkdir -p $(SYNAPSE_CACHE)
+	[ -d $(RESOURCES_DIR) ] || mkdir -p $(RESOURCES_DIR)
+	[ -d $(PRIORS_REFERENCES) ] || mkdir -p $(PRIORS_REFERENCES)
+	[ -d $(OUT_DIR) ] || mkdir -p $(OUT_DIR)
+	[ -f $(PREVIOUS_RELEASE_DIR)/latest_release.tar.gz ] || wget http://brcaexchange.org/backend/downloads/releases/current_release.tar.gz -O $(PREVIOUS_RELEASE_DIR)/latest_release.tar.gz
+
+.ONESHELL:
+setup-lovd: ## setting up LOVD data via pipeline machine, if not running on pipeline machine, as the access is IP restricted. Requires access to main pipeline machine	
+	if [ `hostname -s` != "brca-pipeline" ]; then
+		mkdir -p $(OUT_DIR)/LOVD
+		ssh pipeline@52.151.62.136 curl https://databases.lovd.nl/shared/export/BRCA > $(OUT_DIR)/LOVD/BRCA.txt
+	fi
+
+# TODO: docker access within docker doesn't work yet on Mac OS X
+DOCKER_RUN_CMD = docker run -it --rm -u `id -u ${USER}`:`stat -c '%g' /var/run/docker.sock` \
```

Comment:
Minor note, but if we plan to support OS X, the formatting flag for `stat` is `-f` instead of `-c`.

Reply:
2 cents: Trying to support two platforms has a cost. I go back and forth with this (spent about 2 hours last night fighting with OSX so I can run tensorflow while on a plane, which requires python 3.5, which causes pain on OSX....poof...2 hours. I haven't tried it on a Mac but in theory if the docker daemon is run in the first linux container (russian doll style) it might work. That said ultimately I think we should try and get rid of dind. In other projects I work on it can cause problems due to host daemon incompatibility issues - sometimes subtle. Ideally we have a fairly basic parent script that then calls a series of dockerized ingest and transform containers.

Reply:
Yes, it has a cost. However, it seems not an excessive one: instead of using `stat` we can just use a linux compatible `gstat` version on mac, same with `make` and `gmake`. All a dev has to do is to brew install these. The docker-in-docker I still have to figure out. But I don't consider this to be part of this PR, because here we only care about generating new data releases, which happen on Linux machines.